connectors/src/connectors/slack/bot.ts

```diff
@@ -0,0 +1,188 @@
+    where: {
+      slackTeamId: slackTeamId,
+    },
+  });
+
+  if (slackConfigurations.length === 0) {
+    return new Err(
+      new Error(
+        `Failed to find slack configuration for Slack team id: ${slackTeamId}`
+      )
+    );
+  }
+  const slackConfig = slackConfigurations[0];
+  if (!slackConfig) {
+    return new Err(
+      new Error(
+        `Failed to find slack configuration for Slack team id: ${slackTeamId}`
+      )
+    );
+  }
+  const connector = await Connector.findByPk(slackConfig.connectorId);
+  if (!connector) {
+    return new Err(new Error("Failed to find connector"));
+  }
+  const res = await botAnswerMessage(
+    message,
+    slackTeamId,
+    slackChannel,
+    slackUserId,
+    slackMessageTs,
+    connector
+  );
+  if (res.isErr()) {
+    logger.error(
+      {
+        error: res.error,
+        slackTeamId: slackTeamId,
+        slackChannel: slackChannel,
+        slackUserId: slackUserId,
+        slackMessageTs: slackMessageTs,
+      },
+      "Failed answering to Slack Chat Bot message"
+    );
+    const accessToken = await getAccessToken(connector.connectionId);
+    const slackClient = getSlackClient(accessToken);
+    await slackClient.chat.postMessage({
+      channel: slackChannel,
+      text: `An error occured. Our team has been notified and will work on it as soon as possible.`,
+      thread_ts: slackMessageTs,
+    });
+  }
+
+  return res;
+}
+
+async function botAnswerMessage(
+  message: string,
+  slackTeamId: string,
+  slackChannel: string,
+  slackUserId: string,
+  slackMessageTs: string,
+  connector: Connector
+): Promise<Result<ChatSessionUpdateEvent, Error>> {
+  const { DUST_API = "https://dust.tt" } = process.env;
+
+  const slackChatBotMessage = await SlackChatBotMessage.create({
+    connectorId: connector.id,
+    message: message,
+    slackUserId: slackUserId,
+    slackEmail: "",
+    slackUserName: "",
+    channelId: slackChannel,
+    messageTs: slackMessageTs,
+  });
+
+  const accessToken = await getAccessToken(connector.connectionId);
+  const slackClient = getSlackClient(accessToken);
+  const slackUserInfo = await slackClient.users.info({
+    user: slackUserId,
+  });
+
+  if (!slackUserInfo.ok || !slackUserInfo.user) {
+    throw new Error(`Failed to get user info: ${slackUserInfo.error}`);
+  }
+  const slackUser = slackUserInfo.user;
+  if (slackUser.profile?.email) {
+    slackChatBotMessage.slackEmail = slackUser.profile.email;
+  }
+  if (slackUser.profile?.display_name) {
+    slackChatBotMessage.slackUserName = slackUser.profile.display_name;
+  }
+  await slackChatBotMessage.save();
+  const c = new DustAPI({
+    workspaceId: connector.workspaceId,
+    apiKey: connector.workspaceAPIKey,
+  });
+  const mainMessage = await slackClient.chat.postMessage({
+    channel: slackChannel,
+    text: "_I am thinking..._",
+    thread_ts: slackMessageTs,
+    mrkdwn: true,
+  });
+
+  const matches = message.match(/<@[A-Z-0-9]+>/g);
```

Comment:
Can you add a comment about exactly what you're doing here. I think I have the idea but would be super useful to have a comment that covers the two cases :pray: